src/lein_monolith/plugin.clj

```diff
@@ -103,16 +103,38 @@
 (defn build-inherited-profiles
   "Returns a map from profile keys to inherited profile maps."
   [monolith subproject]
-  (reduce
-    (fn [acc [key config]]
-      (let [profile (some-> (choose-inheritance-source monolith config)
-                            (inherited-profile subproject config)
-                            (maybe-mark-leaky config))]
-        (if profile
-          (assoc acc key profile)
-          acc)))
-    nil
-    profile-config))
+  (when (:monolith/inherit subproject)
```

Comment:
I think this is incorrect (and the current tool has a bug) - we also need to check the `:monolith/inherit-raw`, `:monolith/inherit-leaky`, and `:monolith/inherit-leaky-raw` keys. That said, why do we need this guard at all? Shouldn't this return an empty map if none of the subproject keys are set?

Reply:
> Shouldn't this return an empty map if none of the subproject keys are set?

Thanks for calling this out, I didn't look deeply enough into the implementation here and just biased toward guarding against building these profiles 🙈